### ISSUE TITLE
BERT: Cannot get the reported accuracy with the master branch

### ISSUE THREAD

**Issue by TaoLv** [dmlc/gluon-nlp#1127]
Cannot the reported accuracy with the master branch of GluonNLP and a nightly build of MXNet.

Take MRPC fine-tuning as an example:

```
(mxnet) [lvtao@mlt-gpu200 bert]$ python finetune_classifier.py --task_name MRPC --batch_size 32 --epochs 3 --gpu 0 --lr 2e-5
INFO:root:22:47:22 Namespace(accumulate=None, batch_size=32, bert_dataset='book_corpus_wiki_en_uncased', bert_model='bert_12_768_12', dev_batch_size=8, dtype='float32', early_stop=None, epochs=3, epsilon=1e-06, gpu=0, log_interval=10, lr=2e-05, max_len=128, model_parameters=None, only_inference=False, optimizer='bertadam', output_dir='./output_dir', pretrained_bert_parameters=None, seed=2, task_name='MRPC', training_steps=None, warmup_ratio=0.1)
[22:47:24] src/base.cc:51: Upgrade advisory: this mxnet has been built against cuda library version 9020, which is older than the oldest version tested by CI (10000).  Set MXNET_CUDA_LIB_CHECKING=0 to quiet this warning.
INFO:root:22:47:28 processing dataset...
INFO:root:22:47:30 Now we are doing BERT classification training on gpu(0)!
INFO:root:22:47:30 training steps=343
INFO:root:22:47:33 [Epoch 1 Batch 10/119] loss=0.6328, lr=0.0000053, metrics:accuracy:0.6844,f1:0.8112
INFO:root:22:47:34 [Epoch 1 Batch 20/119] loss=0.6421, lr=0.0000112, metrics:accuracy:0.6693,f1:0.7937
INFO:root:22:47:36 [Epoch 1 Batch 30/119] loss=0.6187, lr=0.0000171, metrics:accuracy:0.6786,f1:0.8026
INFO:root:22:47:38 [Epoch 1 Batch 40/119] loss=0.5839, lr=0.0000197, metrics:accuracy:0.6881,f1:0.8085
INFO:root:22:47:39 [Epoch 1 Batch 50/119] loss=0.6971, lr=0.0000190, metrics:accuracy:0.6719,f1:0.7961
INFO:root:22:47:41 [Epoch 1 Batch 60/119] loss=0.6307, lr=0.0000184, metrics:accuracy:0.6696,f1:0.7929
INFO:root:22:47:42 [Epoch 1 Batch 70/119] loss=0.6842, lr=0.0000177, metrics:accuracy:0.6565,f1:0.7792
INFO:root:22:47:44 [Epoch 1 Batch 80/119] loss=0.6355, lr=0.0000171, metrics:accuracy:0.6540,f1:0.7765
INFO:root:22:47:45 [Epoch 1 Batch 90/119] loss=0.6826, lr=0.0000164, metrics:accuracy:0.6540,f1:0.7782
INFO:root:22:47:47 [Epoch 1 Batch 100/119] loss=0.6138, lr=0.0000158, metrics:accuracy:0.6595,f1:0.7838
INFO:root:22:47:49 [Epoch 1 Batch 110/119] loss=0.6562, lr=0.0000151, metrics:accuracy:0.6583,f1:0.7840
INFO:root:22:47:50 Now we are doing evaluation on dev with gpu(0).
INFO:root:22:47:51 [Batch 10/51] loss=0.6115, metrics:accuracy:0.7000,f1:0.8235
INFO:root:22:47:51 [Batch 20/51] loss=0.6306, metrics:accuracy:0.6875,f1:0.8148
INFO:root:22:47:51 [Batch 30/51] loss=0.6021, metrics:accuracy:0.6958,f1:0.8206
INFO:root:22:47:51 [Batch 40/51] loss=0.6307, metrics:accuracy:0.6906,f1:0.8170
INFO:root:22:47:52 [Batch 50/51] loss=0.6401, metrics:accuracy:0.6850,f1:0.8131
INFO:root:22:47:52 validation metrics:accuracy:0.6838,f1:0.8122
INFO:root:22:47:52 Time cost=1.50s, throughput=272.63 samples/s
INFO:root:22:47:56 params saved in: ./output_dir/model_bert_MRPC_0.params
INFO:root:22:47:56 Time cost=26.15s
INFO:root:22:47:58 [Epoch 2 Batch 10/119] loss=0.6394, lr=0.0000139, metrics:accuracy:0.6632,f1:0.7975
INFO:root:22:48:00 [Epoch 2 Batch 20/119] loss=0.6355, lr=0.0000133, metrics:accuracy:0.6710,f1:0.8031
INFO:root:22:48:01 [Epoch 2 Batch 30/119] loss=0.6387, lr=0.0000126, metrics:accuracy:0.6692,f1:0.8018
INFO:root:22:48:03 [Epoch 2 Batch 40/119] loss=0.6460, lr=0.0000120, metrics:accuracy:0.6659,f1:0.7994
INFO:root:22:48:04 [Epoch 2 Batch 50/119] loss=0.6451, lr=0.0000113, metrics:accuracy:0.6665,f1:0.7998
INFO:root:22:48:06 [Epoch 2 Batch 60/119] loss=0.6128, lr=0.0000107, metrics:accuracy:0.6677,f1:0.8008
INFO:root:22:48:08 [Epoch 2 Batch 70/119] loss=0.6269, lr=0.0000100, metrics:accuracy:0.6700,f1:0.8024
INFO:root:22:48:09 [Epoch 2 Batch 80/119] loss=0.6391, lr=0.0000094, metrics:accuracy:0.6691,f1:0.8017
INFO:root:22:48:11 [Epoch 2 Batch 90/119] loss=0.6051, lr=0.0000087, metrics:accuracy:0.6715,f1:0.8035
INFO:root:22:48:12 [Epoch 2 Batch 100/119] loss=0.5598, lr=0.0000081, metrics:accuracy:0.6768,f1:0.8070
INFO:root:22:48:14 [Epoch 2 Batch 110/119] loss=0.6328, lr=0.0000074, metrics:accuracy:0.6781,f1:0.8058
INFO:root:22:48:15 Now we are doing evaluation on dev with gpu(0).
INFO:root:22:48:16 [Batch 10/51] loss=0.5951, metrics:accuracy:0.7125,f1:0.8217
INFO:root:22:48:16 [Batch 20/51] loss=0.6593, metrics:accuracy:0.6750,f1:0.7984
INFO:root:22:48:16 [Batch 30/51] loss=0.6350, metrics:accuracy:0.6792,f1:0.8021
INFO:root:22:48:16 [Batch 40/51] loss=0.6661, metrics:accuracy:0.6687,f1:0.7954
INFO:root:22:48:16 [Batch 50/51] loss=0.6437, metrics:accuracy:0.6725,f1:0.7969
INFO:root:22:48:16 validation metrics:accuracy:0.6691,f1:0.7939
INFO:root:22:48:16 Time cost=1.21s, throughput=338.03 samples/s
INFO:root:22:48:21 params saved in: ./output_dir/model_bert_MRPC_1.params
INFO:root:22:48:21 Time cost=24.79s
INFO:root:22:48:23 [Epoch 3 Batch 10/119] loss=0.6267, lr=0.0000062, metrics:accuracy:0.6869,f1:0.8056
INFO:root:22:48:24 [Epoch 3 Batch 20/119] loss=0.5999, lr=0.0000056, metrics:accuracy:0.6923,f1:0.8068
INFO:root:22:48:26 [Epoch 3 Batch 30/119] loss=0.5741, lr=0.0000049, metrics:accuracy:0.7023,f1:0.8085
INFO:root:22:48:28 [Epoch 3 Batch 40/119] loss=0.5909, lr=0.0000043, metrics:accuracy:0.7073,f1:0.8122
INFO:root:22:48:29 [Epoch 3 Batch 50/119] loss=0.5718, lr=0.0000036, metrics:accuracy:0.7042,f1:0.8101
INFO:root:22:48:31 [Epoch 3 Batch 60/119] loss=0.5747, lr=0.0000030, metrics:accuracy:0.7084,f1:0.8130
INFO:root:22:48:32 [Epoch 3 Batch 70/119] loss=0.6543, lr=0.0000023, metrics:accuracy:0.6932,f1:0.7982
INFO:root:22:48:34 [Epoch 3 Batch 80/119] loss=0.5972, lr=0.0000017, metrics:accuracy:0.6906,f1:0.7954
INFO:root:22:48:35 [Epoch 3 Batch 90/119] loss=0.5921, lr=0.0000010, metrics:accuracy:0.6917,f1:0.7971
INFO:root:22:48:37 [Epoch 3 Batch 100/119] loss=0.5446, lr=0.0000004, metrics:accuracy:0.6964,f1:0.8009
INFO:root:22:48:38 Finish training step: 343
INFO:root:22:48:38 Now we are doing evaluation on dev with gpu(0).
INFO:root:22:48:38 [Batch 10/51] loss=0.5878, metrics:accuracy:0.6875,f1:0.7934
INFO:root:22:48:38 [Batch 20/51] loss=0.6297, metrics:accuracy:0.6562,f1:0.7755
INFO:root:22:48:38 [Batch 30/51] loss=0.6211, metrics:accuracy:0.6583,f1:0.7784
INFO:root:22:48:39 [Batch 40/51] loss=0.6200, metrics:accuracy:0.6594,f1:0.7789
INFO:root:22:48:39 [Batch 50/51] loss=0.5931, metrics:accuracy:0.6675,f1:0.7837
INFO:root:22:48:39 validation metrics:accuracy:0.6667,f1:0.7821
INFO:root:22:48:39 Time cost=1.27s, throughput=321.60 samples/s
INFO:root:22:48:43 params saved in: ./output_dir/model_bert_MRPC_2.params
INFO:root:22:48:43 Time cost=22.36s
INFO:root:22:48:47 Best model at epoch 0. Validation metrics:accuracy:0.6838,f1:0.8122
INFO:root:22:48:47 Now we are doing testing on test with gpu(0).
INFO:root:22:48:52 Time cost=4.64s, throughput=372.78 samples/s
```

GluonNLP is installed from source code. Commit id: aa663284c9c94c0792f3d0ac901e1af2ef81e1fb
MXNet nightly build: mxnet-cu92 (1.6.0b20200118)

**Comment by zburning** [comment]
Hi @TaoLv , within GluonNLP built from master and mxnet-cu92(1.6.0b20200126) and mxnet-cu100 built from source(branch 1.6.x), I can get reasonable results(accuracy:0.8480, f1:0.8927) using your command. I'll switch to your mxnet version but maybe you can first try one of the above two versions?

**Comment by TaoLv** [comment]
May I know how did you install GluonNLP from source code and what's you pre-trained model? I will try to find a clear environment and do that again.

**Comment by zburning** [comment]
@TaoLv, Hi I can reproduce your result by switching to the commit [aa66328](https://github.com/dmlc/gluon-nlp/commit/aa663284c9c94c0792f3d0ac901e1af2ef81e1fb),  I think maybe something wrong with Commit:[e88d55e77e55b8e562881ed9d12160053810961b](https://github.com/dmlc/gluon-nlp/commit/e88d55e77e55b8e562881ed9d12160053810961b). Currently, you can reset to commit:[5fe8d7b](https://github.com/dmlc/gluon-nlp/commit/5fe8d7b7f5770d82e4f3bfb489a67999d9442ce6) to get the matched result.

**Comment by TaoLv** [comment]
@zburning Thank you for the confirmation. Also @eric-haibin-lin .
